frontend/cypress/e2e/filtro.spec.js

```diff
@@ -0,0 +1,24 @@
+describe('Filtro de Clientes - Mantém Filtros ao Voltar dos Detalhes', () => {
+    beforeEach(() => {
+      cy.visit('http://localhost:3000');
+  
+      cy.get('[data-testid="size-filter"]').select('Medium');
+      cy.get('[data-testid="industry-filter"]').select('Technology');
+  
+      cy.get('[data-testid="table"]').should('exist');
+      cy.get('tbody tr').should('have.length.greaterThan', 0);
+    });
+  
+    it('deve manter os filtros aplicados após voltar da página de detalhes do cliente', () => {
+      cy.get('tbody tr').first().find('button[aria-label^="View company"]').click();
+  
+      cy.get('h2').contains('Customer Details');
```

Comment:
Aqui poderia ser:

```suggestion
      cy.contains('h2', 'Customer Details').should('be.visible');
```

Perceba também a verificação pela visibilidade do elemento.

Falo sobre a importância de tal verificação no vídeo chamado [Vamos conversar sobre assertions](https://youtube.com/live/NBCILY2qsxQ?feature=share)